batchspawner/batchspawner.py

```diff
@@ -775,4 +777,106 @@ def state_gethost(self):
         self.log.error("Spawner unable to match host addr in job {0} with status {1}".format(self.job_id, self.job_status))
         return
 
+
+class OARSpawner(UserEnvMixin, BatchSpawnerBase):
+    batch_script = Unicode("""#!/bin/sh
+#OAR -n jupyterhub-singleuser-{{username}}
+#OAR -l nodes=1{% if nprocs     %}/core={{nprocs}}{% endif %}{% if runtime    %},walltime={{runtime}}{% endif %}
+{% if queue      %}#OAR -q {{queue}}{% endif %}
+{% if memory     %}#OAR -p "memnode>={{memory}}"{% endif %}
+{% if options    %}#OAR {{options}}{% endif %}
+
+{{prologue}}
+{{cmd}}
+{{epilogue}}
+""").tag(config=True)
+
+    # use a script to submit to batch queueing system instead of piping it in
+    script_mode = stat.S_IRUSR | stat.S_IWUSR | stat.S_IRGRP | stat.S_IROTH | stat.S_IXUSR | stat.S_IXGRP | stat.S_IXOTH #755 by default
+
+    script_home = os.path.join(os.environ.get("HOME", "/tmp"), "oar_jobs")
+    if not os.path.exists(script_home):
+        os.makedirs(script_home)
```

Comment:
I guess `script_home` could be made configurable, too?

And maybe the makedirs could be made when it's used in `run_command` - as it is just importing the module will have side-effects of making that directory, actually for everyone who uses batchspawner, not just those who use OARSpawner.   It also has the possibility of making errors at import-time.

Reply:
Indeed ... I will correct this.